android/app/src/main/java/com/sample/MainActivity.kt

```diff
@@ -32,21 +35,67 @@ class MainActivity : ReactActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
+        val reactNativeHost = reactNativeHost
+        if (reactNativeHost is TestAppReactNativeHost) {
+            reactNativeHost.currentActivity = this
```

Comment:
Setting `currentActivity` is taken care of by `ReactActivity` that is extended by `MainActivity`. Is there a particular reason to set here explicitly?

Reply:
`ReactActivity.onResume -> ReactActivityDelegate.onResume -> ReactInstanceManager.onHostResume`:

```
  @ThreadConfined(UI)
  public void onHostResume(Activity activity) {
    UiThreadUtil.assertOnUiThread();

    mCurrentActivity = activity;
    ...
}
```

Reply:
Yes, I get NPEs otherwise. I don't think `onHostResume` gets called on initialisation. Not on my device at least.

Reply:
It doesn't set the `currentActivity` in `ReactActivityDelegate.onResume` because it can't find an instance of `ReactInstanceManager`, that is lazily created whenever you call `reactNativeHost.reactInstanceManager`. The way how we can indirectly initialise it is by calling `reactInstanceManager`. 

I see this working pretty well with another PR of yours for setting runtime info. We can poll the JS executor prop out of `ReactInstanceManager` instead of defining a custom build prop :) 

```
findViewById<TextView>(R.id.runtimeInfo).text =
            reactNativeHost.reactInstanceManager?.jsExecutorName
```

This way we won't need to have `currentActivity`, and to override `createReactInstanceManager` :)

Reply:
Right, I see. In my opinion, we either keep lazy instantiating `ReactInstanceManager` or we do it earlier. I don't want to rely on an implicit behaviour like getting the JS executor name. Let's address this in a follow-up PR?

Reply:
Sure, we can explicitly initialise `ReactInstanceManager` by calling `createReactInstanceManager()` within `TestAppReactNativeHost` for example. Let's look at it in the separate PR.